### ISSUE TITLE
$ in password break install chart

### ISSUE THREAD

**Issue by borgez** [bitnami/charts#2771]
**Which chart**:
postgresql-ha:3.2.9

**Describe the bug**
postgresql-0 cannot start with error in console
/opt/bitnami/scripts/postgresql-repmgr/entrypoint.sh: line 21: An9: unbound variable

**To Reproduce**
1. user password with $
4. publish chart

**Expected behavior**
any password allowed

**Comment by andresbono** [comment]
Hi @borgez, thanks for opening this issue. What is the password parameter that you are using and that triggers the issue?

**Comment by borgez** [comment]
> Hi @borgez, thanks for opening this issue. What is the password parameter that you are using and that triggers the issue?

Hi, i try parameters like this, don't know which password triggers, after change password error gone

```
postgresql-ha:
  postgresql:
    # move to vars in release
    # do not use $
    # do not update without right pass (it regenerate)
    password: Qs&4&$E8RKm&lh!
    repmgrPassword: Qs&4&$E8RKm&lh!
```

**Comment by andresbono** [comment]
Thanks, we could reproduce it now. Your case would be the second one:

#### Set `postgresql.repmgrPassword` with `$`

```
$ helm install postgresqlha bitnami/postgresql-ha --set postgresql.repmgrPassword='Qs&4&Y$An9%Km&lh!'

$ kubectl logs postgresqlha-postgresql-ha-pgpool-7f9dfb8549-wx7ql
...
pgpool 08:45:41.30 INFO  ==> Generating password file for local authentication...
WARNING: pid 85: syntex error in configuration file "/opt/bitnami/pgpool/conf/pgpool.conf"
Unable to get configuration. Exiting...
```

#### Set `postgresql.password` with `$`

```
$ helm install postgresqlha bitnami/postgresql-ha --set postgresql.password='Qs&4&Y$An9%Km&lh!'

$ kubectl logs postgresqlha-postgresql-ha-postgresql-0
/opt/bitnami/scripts/postgresql-repmgr/entrypoint.sh: line 21: An9: unbound variable
```

---

It seems a inherent problem to use bash logic to handle the initialization. Do you have any suggestions to improve this behavior?

**Comment by borgez** [comment]
> It seems a inherent problem to use bash logic to handle the initialization. Do you have any suggestions to improve this behavior?

1) update docs about this behavior or
2) use password as plain text

ps update docs about repmgrPassword is regenerate on each update, it breaks cluster

**Comment by andresbono** [comment]
Hi @borgez, we could identify a posible cause of the issue. We will release a new version of the chart soon

**Comment by marcosbc** [comment]
Hi @borgez, thanks for pinging us. We're tracking the other ticket and expect to check it shortly.

As for this issue, we've fixed it in the release `3.2.10` of this chart. I'm closing this, but if you find any issues please let us know.
